types/config.go

```diff
@@ -0,0 +1,67 @@
+package types
+
+import (
+	"encoding/json"
+)
+
+// VMConfig defines the configuration for the VM.
+// For full documentation see the Rust side:
+// https://docs.rs/cosmwasm-vm/2.2.0-rc.1/cosmwasm_vm/struct.Config.html
+type VMConfig struct {
+	WasmLimits WasmLimits   `json:"wasm_limits"`
+	Cache      CacheOptions `json:"cache"`
+}
+
+// WasmLimits defines limits for static validation of wasm code that is stored on chain.
+// For full documentation see the Rust side:
+// https://docs.rs/cosmwasm-vm/2.2.0-rc.1/cosmwasm_vm/struct.WasmLimits.html
+type WasmLimits struct {
+	InitialMemoryLimit     *uint32 `json:"initial_memory_limit,omitempty"`
+	TableSizeLimit         *uint32 `json:"table_size_limit,omitempty"`
+	MaxImports             *uint32 `json:"max_imports,omitempty"`
+	MaxFunctions           *uint32 `json:"max_functions,omitempty"`
+	MaxFunctionParams      *uint32 `json:"max_function_params,omitempty"`
+	MaxTotalFunctionParams *uint32 `json:"max_total_function_params,omitempty"`
+	MaxFunctionResults     *uint32 `json:"max_function_results,omitempty"`
+}
+
+type CacheOptions struct {
+	BaseDir               string   `json:"base_dir"`
+	AvailableCapabilities []string `json:"available_capabilities"`
+	MemoryCacheSize       Size     `json:"memory_cache_size"`
+	InstanceMemoryLimit   Size     `json:"instance_memory_limit"`
+}
+
+type Size struct{ uint32 }
```

Comment:
How certain are we that 4GiB (2^32) is enough to cover all scenarios?

Reply:
See responses under your previous review comment.